pkg/diagnostics/network/setup.go

```diff
@@ -152,23 +154,23 @@ func (d *NetworkDiagnostic) waitForTestPodAndService(nsList []string) error {
 		}
 	}
 
-	if len(errList) > 0 {
-		numValidPods, totalPods, err := d.getNumOfValidTestPods(nsList, validPhases)
-		if err == nil {
-			// Perform network diagnostic checks if we are able to launch
-			// decent number of test pods (at least 50%)
-			if numValidPods >= (totalPods / 2) {
-				d.res.Warn("DNet3002", nil, fmt.Sprintf("Failed to run some network diags test pods: %v, So some network diagnostic checks may be skipped.", kerrors.NewAggregate(errList)))
+	if totalPods, runningPods, err := d.getCountOfTestPods(nsList); err == nil {
+		// Perform network diagnostic checks if we are able to launch decent number of test pods (at least 50%)
+		if runningPods != totalPods {
+			if runningPods >= (totalPods / 2) {
+				d.res.Warn("DNet3002", nil, fmt.Sprintf("Failed to run some network diags test pods: %d, So some network diagnostic checks may be skipped.", (totalPods-runningPods)))
 				return nil
+			} else {
+				errList = append(errList, fmt.Errorf("Failed to run network diags test pods, failed: %d, total: %d", (totalPods-runningPods), totalPods))
```

Comment:
It would be really nice here if it could report on what happened with the ones that aren't running.